### DISCUSSION TITLE
Open and close panels via shortcut key

### DISCUSSION THREAD

**Discussion by F-Burning** [opentoonz/opentoonz#6805] [Ideas]
When you open a panel with a shortcut, you can't close it with the same key

https://user-images.githubusercontent.com/74098734/113760918-c081cb00-970e-11eb-8074-4534e9bc26a4.mp4

Isn't it more convenient to be able to open and close it with the same shortcut?

Edit: In the same way, when you open a window and expand or collapse it and then close opentoonz, when you reopen it the window does not remember the last configuration, and it reopens by default. For example if I want the File browser windows expanded, I can't because it always opens collapsed.

**Comment by DarrenTAnims** [comment]
I think this is done on purpose. The key shortcut is to open a panel of that type, but you might want multiple, as your video shows. The shortcut isn't for opening just one. However, I can see a benefit for a toggle button to open and close a panel. The question is, if you've got multiple already open, which does it close?

**Comment by F-Burning** [comment]
@DarrenTAnims that's true.

What do you think about an option like this?

![image](https://user-images.githubusercontent.com/74098734/113768800-26268500-9718-11eb-9099-dc2183feb315.png)

If you activate it, all the panels close in the same room, but this option is fine if you only need the panel momentarily without taking up space in the room I think.

**Comment by manongjohn** [comment]
> The question is, if you've got multiple already open, which does it close?

I say it should close the current active one.

I would probably go with ESC key should close the any active floating panel you are on. Use Shortcut to open panel, ESC to close it when done. Lots of other popups close with ESC.

**Comment by F-Burning** [comment]
> I say it should close the current active one.

The problem with this is that then you could not open a new one without closing the one that is already in the room I think.

ESC is the one that convinces me the most 😄 although the option to do it with the same key is tentative.
